### PR TITLE
dev: add basic embeddings QA script

### PR DESCRIPTION
**Pull request by stefanhengl**
This adds the "sg embeddings-qa" command. The command calculates _recall_ for a curated test set (prompts -> relevant context files). The goal is to use recall to assess changes to the embeddings service. The list of prompts will grow over time.

I went back and forth between targeting the embeddings  service or the GraphQL API. I ended up targeting the embeddings service directly because it is simpler and this moment there is no benefit in having the GraphQL API in the middle. Once we mix in document ranks (currently disabled) we should reevaluate. 

 ## Test plan
- run
```
# terminal 1
sg start embeddings

# terminal 2
cd /dev/sg 
go run . embeddings-qa
```
- check outpout

### CODE REVIEW

File: enterprise/cmd/embeddings/qa/context_data.tsv [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Where does sourcegraph convert lang filters to file extensions?	internal/search/zoekt/query.go
```
Comment (camdencheek) [inline enterprise/cmd/embeddings/qa/context_data.tsv:1]: Wouldn't a better file be [this](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@262405666de64ef3778174acedc30add90eb8663/-/blob/internal/search/query/helpers.go?L48-50)? `internal/search/zoekt/query.go` uses that function, but there are a few other places that use it too.

File: enterprise/cmd/embeddings/qa/eval.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+			return errors.Wrap(err, "search failed")
+		}
+
+		merged := append(results.CodeResults, results.TextResults...)
+		fmt.Println("Query:", query)
+		fmt.Println("Results:")
+
+		for i, result := range merged {
+			if result.FileName == relevantFile {
+				recall++
```
Comment (camdencheek) [inline enterprise/cmd/embeddings/qa/eval.go:65]: Should we break in the case that we find a matching file? I think it's possible for multiple chunks from the same file to match, so we could increment recall multiple times for the same query.

Comment (jtibshirani) [reply enterprise/cmd/embeddings/qa/eval.go:65]: +1 I forgot to push this bug fix to the original branch.

**Review by jtibshirani — COMMENTED** [review]
It's great to have this committed so other people can use the script too.

This works well for the following use case: you make a change to embeddings search, and manually run this before and after to compare results. But if we want to run it on CI, it's trickier -- the codebase can change over time, so even if we don't make any change to the embedding service, the recall could change. Maybe for CI, we'll adapt the script to a new static dataset?

Just wanted to get a sense of your plans. This is not blocking at all, I think this command is already useful.

File: enterprise/cmd/embeddings/qa/context_data.tsv (hunk 1 above)
Comment (jtibshirani) [reply enterprise/cmd/embeddings/qa/context_data.tsv:1]: Oops, I agree that one is better! This raises a good point for future iterations of the script -- sometimes several files can be relevant, to different degrees. We may need a more advanced quality metric that is able to capture this.

**Comment by stefanhengl** [comment]
> Just wanted to get a sense of your plans. This is not blocking at all, I think this command is already useful

For CI I want to check with dev-experience. Ideally we could run the script against a snapshot-embedding of the sourcegraph repo.

File: enterprise/cmd/embeddings/qa/eval.go (hunk 2 above)
Comment (stefanhengl) [reply enterprise/cmd/embeddings/qa/eval.go:65]: fixed

File: enterprise/cmd/embeddings/qa/eval.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		}
+
+		merged := append(results.CodeResults, results.TextResults...)
+		fmt.Println("Query:", query)
+		fmt.Println("Results:")
+
+		for i, result := range merged {
+			if result.FileName == relevantFile {
+				// Don't count duplicates.
+				uniqueMatches[result.FileName] = struct{}{}
```
Comment (jtibshirani) [inline enterprise/cmd/embeddings/qa/eval.go:67]: This approach is a bit surprising, doesn't it break in the case that some file is the correct answer for two different queries? I think we could just have a boolean like `foundFile` and then increment the recall later whenever `foundFile` is true.

**Review by jtibshirani — APPROVED** [review]
Left one last comment, other than that it looks good to me!

File: doc/dev/background-information/sg/reference.md [hunk 4]
```diff
@@ -1106,6 +1106,18 @@ Flags:
 * `--feedback`: provide feedback about this command by opening up a GitHub discussion
 * `--format, -f="<value>"`: Output format of list ('markdown', 'plain', 'regexp') (default: markdown)
 
+## sg embeddings-qa
+
+Calculate recall for embeddings.
```
Comment (mrnugget) [inline doc/dev/background-information/sg/reference.md:1111]: Hey! It's me, the guy who asks the dumb questions: can I get a definition for "recall" in this context?

**Pull request merged**